Lib/fontbakery/profiles/cff.py

```diff
@@ -83,70 +93,120 @@ def _check_call_depth(top_dict, private_dict, fd_index=0):
-                    private_dict = font_dict.Private
-                else:
-                    private_dict = None
-                failed = yield from \
-                    _check_call_depth(top_dict, private_dict, fd_index)
-                any_failures = any_failures or failed
-        else:
-            if hasattr(top_dict, 'Private'):
-                private_dict = top_dict.Private
-            else:
-                private_dict = None
-            failed = yield from _check_call_depth(top_dict, private_dict)
-            any_failures = any_failures or failed
+    if cff_analysis.glyphs_exceed_max or cff_analysis.glyphs_recursion_errors:
+        any_failures = True
+        for gn in cff_analysis.glyphs_exceed_max:
+            yield FAIL, Message('max-depth', f'Subroutine call depth exceeded maximum of 10 for glyph "{gn}".')
+        for gn in cff_analysis.glyphs_recursion_errors:
+            yield FAIL, Message('recursion-error', f'Recursion error while decompiling glyph "{gn}".')
 
     if not any_failures:
         yield PASS, 'Maximum call depth not exceeded.'
 
 
 @check(
     id = 'com.adobe.fonts/check/cff2_call_depth',
-    conditions = ['is_cff2'],
+    conditions = ['ttFont', 'is_cff2', 'cff_analysis'],
     rationale = """
         Per "The CFF2 CharString Format", the "Subr nesting, stack limit" is 10.
     """
 )
-def com_adobe_fonts_check_cff2_call_depth(ttFont):
+def com_adobe_fonts_check_cff2_call_depth(cff_analysis):
     """Is the CFF2 subr/gsubr call depth > 10?"""
+
     any_failures = False
-    cff = ttFont['CFF2'].cff
 
-    for top_dict in cff.topDictIndex:
-        for fd_index, font_dict in enumerate(top_dict.FDArray):
-            if hasattr(font_dict, 'Private'):
-                private_dict = font_dict.Private
-            else:
-                private_dict = None
-            failed = yield from \
-                _check_call_depth(top_dict, private_dict, fd_index)
-            any_failures = any_failures or failed
+    if cff_analysis.glyphs_exceed_max or cff_analysis.glyphs_recursion_errors:
+        any_failures = True
+        for gn in cff_analysis.glyphs_exceed_max:
+            yield FAIL, Message('max-depth', f'Subroutine call depth exceeded maximum of 10 for glyph "{gn}".')
+        for gn in cff_analysis.glyphs_recursion_errors:
+            yield FAIL, Message('recursion-error', f'Recursion error while decompiling glyph "{gn}".')
 
     if not any_failures:
         yield PASS, 'Maximum call depth not exceeded.'
+
+
+@check(
+    id = 'com.adobe.fonts/check/cff_deprecated_operators',
+    conditions = ['ttFont', 'is_cff', 'cff_analysis'],
+    rationale = """
+        The 'dotsection' and 'seac' operators (including the use of 'endchar' as
+        implied 'seac') are deprecated in CFF. Adobe recommends repairing any
+        fonts that use them, especially 'seac', which can result in incorrect
+        rendering in some applications.
+    """
```

Comment:
please also include here in the rationale metadata string the remaining info you provided on the pull request description, with a rephrasing of this:
"a rendering issue was discovered in Microsoft Word with a font that makes use of the implied seac. The check treats this as a FAIL. There are no known ill effects of including dotsection, so that check is a WARN."

Reply:
Updated in https://github.com/googlefonts/fontbakery/pull/3033/commits/66b8ecf818c9ec236bdb0fdf75a3bcde982f8b51